### PR TITLE
MUMUP-1952 : Marketplace details cleanup

### PR DESCRIPTION
**Pull request by timlevett**
+ `MarketplaceService` now has `getPortlet(fname)`. This service will use the marketplace cache if there, otherwise it'll call `/portal/web/marketplace/entry/{fname}.json` (yet to be done on /portal side) to get a single entry
+ Adds in mock data for `MyCourses`, including ratings. So `/portal/web/marketplace/entry/MyCourses.json`, `/portal/web/marketplace/MyCourses/getRating` and `/portal/web/marketplace/MyCourses/rating/set/[1-5]` all work.
+ Removes duplicate declaration of `MarketplaceDetailsController`
+ Adds in loading state of details page
http://goo.gl/IK4KW8
+ Adds in error state of details page
http://goo.gl/FYEH5y

### CODE REVIEW

**Comment by jhanstra** [comment]
Awesome, :+1:

File: angularjs-portal-home/src/main/webapp/my-app/marketplace/partials/marketplace-details.html [hunk 1]
```diff
@@ -1,105 +1,106 @@
...
+		<div class="col-xs-12 col-sm-4 col-lg-3 desc-item center">
+			<h3>Categories</h3>
+			<a ng-repeat="category in portlet.categories" href="" ng-click="selectFilter('category',category)" class="category-links">{{::category}}</a>
+		</div>
+		<div class="col-xs-12 col-sm-4 col-lg-3 desc-item center" ng-if="$storage.showKeywordsInMarketplace">
+			<h3>Keywords</h3>
+	    <ul>
+	      <li ng-repeat="keyword in portlet.keywords | orderBy:'toLowerCase()'  track by $index"><a href='apps/search/{{ ::keyword }}'>{{ ::keyword }}</a></li>
+	    </ul>
+		</div>
```
Comment (apetro) [inline angularjs-portal-home/src/main/webapp/my-app/marketplace/partials/marketplace-details.html]: Tabs for whitespace?

**Comment by apetro** [comment]
:+1: ; probably should hold off on merge until `/portal/web/marketplace/entry/{fname}.json` exists portal-side.

File: angularjs-portal-home/src/main/webapp/my-app/marketplace/partials/marketplace-details.html (hunk 1 above)
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/my-app/marketplace/partials/marketplace-details.html]: Didn't change that, but did get rid of an outer div.

**Comment by timlevett** [comment]
Merging now, only breaks if you deep link to marketplace entry, before hitting marketplace.

**Comment by timlevett** [comment]
Portal pull: https://github.com/UW-Madison-DoIT/uPortal/pull/372

**Pull request merged**